test/sql/local/partitioning/year/year_timestamp.test

```diff
@@ -0,0 +1,69 @@
+# name: test/sql/local/partitioning/year/year_timestamp.test
+# group: [year]
+
+require avro
+
+require parquet
+
+require iceberg
+
+require httpfs
+
+require-env DUCKDB_ICEBERG_HAVE_GENERATED_DATA
+
+query III
+select * from ICEBERG_SCAN('data/generated/iceberg/spark-local/default/year_timestamp') ORDER BY partition_col;
+----
+2020-05-15 14:30:45	12345	click
+2021-08-22 09:15:20	67890	purchase
+2022-03-10 11:45:30	54321	view
+2023-01-01 00:00:00	NULL	null_user
+2023-02-15 12:30:45	88888	NULL
+NULL	99999	null_event
+
+query I
+select user_id from ICEBERG_SCAN('data/generated/iceberg/spark-local/default/year_timestamp') 
+WHERE partition_col = NULL::TIMESTAMP
+----
+
+query I
+select user_id from ICEBERG_SCAN('data/generated/iceberg/spark-local/default/year_timestamp') 
+WHERE partition_col IS NULL
+----
+99999
+
+query I
+select user_id from ICEBERG_SCAN('data/generated/iceberg/spark-local/default/year_timestamp') 
+WHERE partition_col IS NOT NULL ORDER BY user_id
+----
+12345
+54321
+67890
+88888
+NULL
+
+# Test filtering by year
+query I
+select user_id from ICEBERG_SCAN('data/generated/iceberg/spark-local/default/year_timestamp') 
```

Comment:
could we perhaps use logging to verify that we successfully skip files by the partition transform?